src/lib.rs

```diff
@@ -69,75 +94,64 @@ pub unsafe extern "C" fn pthread_join(
     native: libc::pthread_t,
     _value: *mut *mut libc::c_void,
 ) -> libc::c_int {
-    if native == MAIN_THREAD_ID {
-        // This is not a valid thread to join on
-        return libc::EINVAL;
-    }
+    let pthread = Box::from_raw(native as *mut PThread);
 
-    let result = ctru_sys::threadJoin(native as ctru_sys::Thread, u64::MAX);
+    let result = ctru_sys::threadJoin(pthread.thread, u64::MAX);
```

Comment:
I might have screwed something up with join... `thread-info` example shows four threads during and after the thread is joined, which seems incorrect.

Reply:
Need to re-test this with latest changes

Reply:
Ok, got to the bottom of this, we need to close the handles we are opening for setting sched params. I will push a fix shortly.